variables.tf

```diff
@@ -0,0 +1,241 @@
+variable "ssh_key_pair" {
+  description = "SSH key pair to be provisioned on the instance"
+  default     = ""
+}
+
+variable "generate_ssh_key_pair" {
+  description = "If true, create a new key pair and save the pem for it to the current working directory"
+  default     = "false"
+}
+
+variable "associate_public_ip_address" {
+  description = "Associate a public IP address with the instance"
+  default     = "true"
+}
+
+variable "ssh_key_pair_path" {
+  description = "Path to where the generated key pairs will be created. Defaults to $${path.cwd}"
+  default     = ""
+}
+
+variable "assign_eip_address" {
+  description = "Assign an Elastic IP address to the instance"
+  default     = "true"
+}
+
+variable "user_data" {
+  description = "Instance user data. Do not pass gzip-compressed data via this argument"
+  default     = ""
+}
+
+variable "instance_type" {
+  description = "The type of the instance"
+  default     = "t2.micro"
+}
+
+variable "vpc_id" {
+  description = "The ID of the VPC that the instance security group belongs to"
+}
+
+variable "security_groups" {
+  description = "List of Security Group IDs allowed to connect to the instance"
+  type        = "list"
+  default     = []
+}
+
+variable "allowed_ports" {
+  type        = "list"
+  description = "List of allowed ingress ports"
+  default     = []
+}
+
+variable "subnet" {
+  description = "VPC Subnet ID the instance is launched in"
+}
+
+variable "namespace" {
+  description = "Namespace (e.g. `cp` or `cloudposse`) - required for `tf_label` module"
+}
+
+variable "stage" {
+  description = "Stage (e.g. `prod`, `dev`, `staging` - required for `tf_label` module"
+}
+
+variable "name" {
+  description = "Name  (e.g. `bastion` or `db`) - required for `tf_label` module"
+}
+
+variable "delimiter" {
+  default = "-"
+}
+
+variable "attributes" {
+  description = "Additional attributes (e.g. `policy` or `role`)"
+  type        = "list"
+  default     = []
+}
+
+variable "tags" {
+  description = "Additional tags"
+  type        = "map"
+  default     = {}
+}
+
+variable "region" {
+  description = "AWS Region the instance is launched in"
+}
+
+variable "availability_zone" {
+  description = "Availability Zone the instance is launched in. If not set, will be launched in the first AZ of the region"
+  default     = ""
+}
+
+variable "ami" {
+  description = "The AMI to use for the instance. By default it is the AMI provided by Amazon with Ubuntu 16.04"
+  default     = ""
+}
+
+variable "ebs_optimized" {
+  description = "Launched EC2 instance will be EBS-optimized"
+  default     = "false"
+}
+
+variable "disable_api_termination" {
+  description = "Enable EC2 Instance Termination Protection"
+  default     = "false"
+}
+
+variable "monitoring" {
+  description = "Launched EC2 instance will have detailed monitoring enabled"
+  default     = "true"
+}
+
+variable "private_ip" {
```

Comment:
is this the same private IP address for all instances?
if we create more than one instance, will it work?
probably need to provide a list of IPs.